irc_server.go

```diff
@@ -149,15 +149,15 @@ func SendIrcNumeric(ctx *IrcContext, code int, args, desc string) error {
 // channel.
 func IrcSendChanInfoAfterJoin(ctx *IrcContext, name, id, topic string, members []string, isGroup bool) {
 	// TODO wrap all these Conn.Write into a function
-	ctx.Conn.Write([]byte(fmt.Sprintf(":%v JOIN #%v\r\n", ctx.Mask(), name)))
+	ctx.Conn.Write([]byte(fmt.Sprintf(":%v JOIN %v\r\n", ctx.Mask(), name)))
 	// RPL_TOPIC
-	SendIrcNumeric(ctx, 332, fmt.Sprintf("%s #%s", ctx.Nick(), name), topic)
+	SendIrcNumeric(ctx, 332, fmt.Sprintf("%s %s", ctx.Nick(), name), topic)
 	// RPL_NAMREPLY
-	SendIrcNumeric(ctx, 353, fmt.Sprintf("%s = #%s", ctx.Nick(), name), strings.Join(ctx.UserIDsToNames(members...), " "))
+	SendIrcNumeric(ctx, 353, fmt.Sprintf("%s = %s", ctx.Nick(), name), strings.Join(ctx.UserIDsToNames(members...), " "))
 	// RPL_ENDOFNAMES
-	SendIrcNumeric(ctx, 366, fmt.Sprintf("%s #%s", ctx.Nick(), name), "End of NAMES list")
+	SendIrcNumeric(ctx, 366, fmt.Sprintf("%s %s", ctx.Nick(), name), "End of NAMES list")
 	ctx.ChanMutex.Lock()
-	ctx.Channels[name] = Channel{Topic: topic, Members: members, ID: id, IsGroup: isGroup}
+	ctx.Channels[name[1:]] = Channel{Topic: topic, Members: members, ID: id, IsGroup: isGroup}
```

Comment:
what about using the full name with prefix here? In this case we would preserve the channel type in the name

Reply:
it does make sense, I'll change it.

Reply:
Done